src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPrintMembers.cs

```diff
@@ -0,0 +1,277 @@
+                    if (printableMembers.IsEmpty)
+                    {
+                        // return base.print(builder);
+                        F.CloseMethod(F.Return(basePrintCall));
+                        return;
+                    }
+                    else
+                    {
+                        // if (base.print(builder))
+                        //     builder.Append(", ")
+                        block!.Add(F.If(basePrintCall, makeAppendString(F, builder, ", ")));
+                    }
+                }
+
+                Debug.Assert(!printableMembers.IsEmpty && block is object);
+
+                for (var i = 0; i < printableMembers.Length; i++)
+                {
+                    // builder.Append(<name>);
+                    // builder.Append(" = ");
+                    // builder.Append((object)<value>);
+                    // builder.Append(", "); // except for last member
+
+                    var member = printableMembers[i];
+                    block.Add(makeAppendString(F, builder, member.Name));
+                    block.Add(makeAppendString(F, builder, " = "));
+
+                    var value = member.Kind switch
+                    {
+                        SymbolKind.Field => F.Field(F.This(), (FieldSymbol)member),
+                        SymbolKind.Property => F.Property(F.This(), (PropertySymbol)member),
+                        _ => throw ExceptionUtilities.UnexpectedValue(member.Kind)
+                    };
+
+                    block.Add(F.ExpressionStatement(
+                        F.Call(receiver: builder,
+                            F.WellKnownMethod(WellKnownMember.System_Text_StringBuilder__AppendObject),
+                            F.Convert(F.SpecialType(SpecialType.System_Object), value))));
+
+                    if (i < printableMembers.Length - 1)
+                    {
+                        block.Add(makeAppendString(F, builder, ", "));
+                    }
+                }
+
+                block.Add(F.Return(F.Literal(true)));
+
+                F.CloseMethod(F.Block(block.ToImmutableAndFree()));
+            }
+            catch (SyntheticBoundNodeFactory.MissingPredefinedMember ex)
+            {
+                diagnostics.Add(ex.Diagnostic);
+                F.CloseMethod(F.ThrowNull());
+            }
+
+            static BoundStatement makeAppendString(SyntheticBoundNodeFactory F, BoundParameter builder, string value)
+            {
+                return F.ExpressionStatement(F.Call(receiver: builder, F.WellKnownMethod(WellKnownMember.System_Text_StringBuilder__AppendString), F.StringLiteral(value)));
+            }
+        }
+
+        internal static MethodSymbol? FindValidPrintMembersMethod(TypeSymbol containingType, CSharpCompilation compilation)
+        {
+            if (containingType.IsErrorType() || containingType.IsObjectType())
```

Comment:
>containingType.IsErrorType() [](start = 16, length = 28)

When is this true?

Reply:
This is reached in the `record C2: Error;` scenario.
To find the print method in `Error` we need to know what accessibility to look for, so we need to look for the base type (null on error type though).
For clone method, we always know its accessibility (public) so we didn't hit this problem.

---
In reply to: [469706606](https://github.com/dotnet/roslyn/pull/46616#discussion_r469706606) [](ancestors = 469706606)

Reply:
If it's only the base type we're concerned with, consider checking `BaseTypeNoUseSiteDiagnostics is null` explicitly, since that would cover more potential cases. But it's not clear we need to handle `Accessibility.Private` anyway.

---
In reply to: [470018333](https://github.com/dotnet/roslyn/pull/46616#discussion_r470018333) [](ancestors = 470018333,469706606)